tests/test_events.py

```diff
@@ -12,7 +12,7 @@
 from website.addons.base import signals
 from framework.auth import Auth
 from tests import factories
-from tests.base import OsfTestCase
+from tests.base import OsfTestCase, NotificationTestCase
```

Comment:
Ah, okay! No wonder this PR is so unwieldy: it includes #5452 (the remake of #5377), as I now see in the PR description. I'm going to pause reviewing here until #5452 is merged into `develop` and this PR is rebased. Alternatively, you could make a new PR against the branch in #5452, `leb2dg:feature/global-notification-settings`. It's really distracting to try to review a PR that also includes another PR.